### PR TITLE
TDR-2816 - Add file properties for the descriptive metadata

### PR DESCRIPTION
**Pull request by vimleshtna**
- Add former_reference_department and creating_body properties
- Add or update display properties for file_name_translation, former_reference_department and creating_body properties
- Add file property values for the Language properties

### CODE REVIEW

File: lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+       ('creating_body', 'PropertyType', 'Descriptive', 'text'),
+       ('creating_body', 'Active', 'false', 'boolean'),
+       ('creating_body', 'Datatype', 'text', 'text'),
+       ('creating_body', 'Editable', 'true', 'boolean'),
+       ('creating_body', 'MultiValue', 'false', 'boolean'),
+       ('creating_body', 'Ordinal', '35', 'integer'),
+       ('creating_body', 'Summary', 'Creating body', 'text'),
+       ('end_date', 'Summary', 'Date of the record', 'text');
+
+INSERT INTO "FilePropertyValues" ("PropertyName", "PropertyValue", "Default", "Dependencies", "SecondaryValue")
```
Comment (TomJKing) [inline lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql:52]: Why are these being added? The `Language` property should only have `English` and `Welsh` values. 

Also for the values they should be single values, eg "Arabic", "English". Not a single value for multiple languages, eg "Arabic and English".

File: lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        'Add where the record is inherited from if it is different to the transferring body.', 'text'),
+       ('creating_body', 'ComponentType', 'select', 'text'),
+       ('creating_body', 'PropertyType', 'Descriptive', 'text'),
+       ('creating_body', 'Active', 'false', 'boolean'),
+       ('creating_body', 'Datatype', 'text', 'text'),
+       ('creating_body', 'Editable', 'true', 'boolean'),
+       ('creating_body', 'MultiValue', 'false', 'boolean'),
+       ('creating_body', 'Ordinal', '35', 'integer'),
+       ('creating_body', 'Summary', 'Creating body', 'text'),
+       ('end_date', 'Summary', 'Date of the record', 'text');
```
Comment (TomJKing) [inline lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql:50]: Was the `end_date` property meant to be updated as well?

**Comment by thanhz** [comment]
:bell: Just a reminder to increment the script number as I've just merged a new script

File: lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql [hunk 3]
```diff
@@ -0,0 +1,253 @@
+-- Add former_reference_department and creating_body properties
+-- Add or update display properties for file_name_translation, former_reference_department and creating_body properties
+-- Add file property values for the Language
+
+INSERT INTO "FileProperty" ("Name", "Description", "FullName", "PropertyType", "Datatype", "Editable", "MultiValue",
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql:5]: I wasn't sure about the export. I have added them, thanks for raising this point.

File: lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql (hunk 1 above)
Comment (vimleshtna) [reply lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql:52]: Oh, sorry, I shouldn't have added those values. I have removed them now. Thanks

File: lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql (hunk 2 above)
Comment (vimleshtna) [reply lambda/src/main/resources/db/migration/V111__add_descriptive_file_properties.sql:50]: We are using the `Summary` on the additional metadata start page and it was missing for the `end_date`, so I have added it.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**